ahk_x11 error:
```
Error target ahk_x11 failed to compile:
Showing last frame. Use --error-trace for full trace.

In src/run/display/x11.cr:270:26

 270 | record_fd.not_nil!.wait_readable
                          ^------------
Error: undefined method 'wait_readable' for IO::FileDescriptor

make[1]: *** [Makefile:42: bin/ahk_x11.dev] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:32: bin/ahk_x11] Error 2
```
ameba error:
```
src/ameba/ast/variabling/argument.cr:1:1 [Correctable]
[W] Lint/Formatting: Use built-in formatter to format this source
> module Ameba::AST
  ^

Finished in 621.63 milliseconds
252 inspected, 1 failure
make: *** [Makefile:13: lint] Error 1
```
rtfm
```
Error target rtfm failed to compile:
Module validation failed: !dbg attachment points at wrong subprogram for function
!4315 = distinct !DISubprogram(name: "main", linkageName: "main", scope: !19, file: !19, line: 1, type: !5, scopeLine: 1, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptim>
ptr @main
  %status = alloca i32, align 4, !dbg !4316
!4316 = !DILocation(line: 8, column: 1, scope: !4317)
!4317 = distinct !DISubprogram(name: "main", linkageName: "main", scope: !4318, file: !4318, line: 8, type: !5, scopeLine: 8, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagO>
!4317 = distinct !DISubprogram(name: "main", linkageName: "main", scope: !4318, file: !4318, line: 8, type: !5, scopeLine: 8, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagO>
 (Exception)
  from /build/source/src/llvm/module.cr:83:9 in 'codegen'
  from /build/source/src/compiler/crystal/compiler.cr:220:15 in 'compile'
  from /build/source/src/compiler/crystal/command.cr:359:5 in 'run'
  from /build/source/src/compiler/crystal.cr:11:1 in '__crystal_main'
  from /build/source/src/crystal/main.cr:118:5 in 'main'
  from /build/source/src/crystal/system/unix/main.cr:10:3 in 'main'
  from /nix/store/h7zcxabfxa7v5xdna45y2hplj31ncf8a-glibc-2.40-36/lib/libc.so.6 in '??'
  from /nix/store/h7zcxabfxa7v5xdna45y2hplj31ncf8a-glibc-2.40-36/lib/libc.so.6 in '__libc_start_main'
  from /nix/store/2dvxh0p8mbaz7hmv2hl20cn9s81q3rys-crystal-1.15.0/bin/crystal in '_start'
  from ???
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues

make: *** [Makefile:17: rtfm] Error 1
```

--- a/pkgs/development/compilers/crystal/default.nix
+++ b/pkgs/development/compilers/crystal/default.nix
@@ -335,6 +335,13 @@ rec {
     doCheck = false; # Some compiler spec problems on x86-64_linux with the .0 release
   };
 
+  crystal_1_15 = generic {
+    version = "1.15.1";
+    sha256 = "sha256-L/Q8yZdDq/wn4kJ+zpLfi4pxznAtgjxTCbLnEiCC2K0=";
+    binary = binaryCrystal_1_10;
+    llvmPackages = llvmPackages_18;
+    doCheck = false;
+  };
 
-  crystal = crystal_1_14;
+  crystal = crystal_1_15;
 }

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5764,6 +5764,7 @@ with pkgs;
     crystal_1_11
     crystal_1_12
     crystal_1_14
+    crystal_1_15
     crystal;
 
   crystalline = callPackage ../development/tools/language-servers/crystalline {

